crud/src/main/java/com/redhat/lightblue/rest/crud/AbstractCrudResource.java

```diff
@@ -111,7 +122,7 @@ public Response getSearchesForEntity(@PathParam("entity") String entity,
         }
         CallStatus st=new FindCommand(freq.getEntityVersion().getEntity(),
                                       freq.getEntityVersion().getVersion(),
-                                      freq.toJson().toString()).run();
+                                      freq.toJson().toString(), metrics).run();
```

Comment:
I would prefer starting metric tracking at REST layer and then passing along the ctx to mediator for all requests (including bulk). One benefit that was seen by tracking metrics at mediator level is that we get more insights on errors which occur at mediator level. If we track everything at REST level, it just marks mediator errors as completed requests, instead of error requests.

So RequestMetrics will get passed on along with each request to commands, commands will initialize the context and then pass ctx to mediator for further tracking at mediator level (error tracking and marking request as completed).

I was thinking of marking request completion at REST level, but then that would not be able to handle Bulk requests. So request ending should have to be handled at mediator level itself.